src/docs/asciidoc/modules/ROOT/pages/chapter3.adoc

```diff
@@ -209,21 +230,26 @@ To provide default data, create `src/main/resources/data.sql` and add a couple o
 [source,sql]
 .src/main/resources/data.sql
 ----
-insert into blog (name) values ('First');
-insert into blog (name) values ('Second');
+insert into blog (id, name) values (1, 'First');
+insert into blog (id, name) values (2, 'Second');
 ----
 
-Start your application with `mvn spring-boot:run` (or right-click -> "`Run in your IDE`") and you should see this default data show up in your logs.
+And add the following property to `src/main/resources/application.properties` so H2 initializes properly.
+
+[source,properties]
+----
+spring.jpa.defer-datasource-initialization=true
```

Comment:
Depending on the audience, they may not understand why this is necessary. 
Perhaps add something along the lines of:
> Because data is added via a script `data.sql`, H2 initialization is necessary to align with that of Flyway and Liquibase to manage our database changelog

[source](https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.5-Release-Notes#hibernate-and-datasql)

Reply:
Good point. I'll rework this to add more of an explanation.